### ISSUE TITLE
Rust Analyzer fails to read metadata. (os error 5)

### ISSUE THREAD

**Issue by ClayCore** [rust-analyzer/rust-analyzer#4745]
After installing the `rust-analyzer` extension for vscode, opening any rust project results in the following error.

```[ERROR rust_analyzer::main_loop] failed to load workspace: Failed to read Cargo metadata from Cargo.toml file s:\wplace\rust\vst-test\Cargo.toml: Failed to run `cargo metadata --manifest-path s:\wplace\rust\vst-test\Cargo.toml`: Access denied. (os error 5): Access denied. (os error 5)```

OS: Windows 10 Enterprise LTSC
VSCode Version: 1.45.1

I made sure that cargo is in PATH and running `cargo metadata` inside any project folders does work.

**Comment by lnicola** [comment]
That's an Access Denied error. Does `cargo metadata --manifest-path s:\wplace\rust\vst-test\Cargo.toml` work if you run it from the command line? What kind of drive is `S:`?

**Comment by ClayCore** [comment]
> 
> 
> That's an Access Denied error. Does `cargo metadata --manifest-path s:\wplace\rust\vst-test\Cargo.toml` work if you run it from the command line? What kind of drive is `S:`?

Yes, running that command does work. `S:\` is just a regular NTFS drive partition. The only quirk is the fact that the `wplace` folder is a folder I'm sharing on my network, so other machines can access it.

**Comment by lnicola** [comment]
Yeah, that happened to others, too. You can leave `%USERPROFILE%/.cargo/bin` in `PATH`, it was added by `rustup` and it's useful if you want to compile code from a terminal. We use `CARGO`, but you might be confusing it with `CARGO_HOME` which indeed should point to a directory (but is optional).
